sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2669,6 +2667,47 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         ctx.EXTENDED != null)
     }
 
+  /**
+   * When resolving TBLPROPERTIES, please let the `location` be None and `isOptionClause` be false
+   */
+  private def cleanTableProperties(
```

Comment:
We can have 2 methods
```
def cleanTableProperties {
  // check reserved properties
}

def cleanOptions(...): (Map[String, String], Option[String]) {
  val cleaned = cleanTableProperties(options)
  var path: Option[String] = None
  val pathRemoved = cleaned.filter {
    case if it's path =>
      if (location.isDefined) fail else ...
  }
  (pathRemoved, location)
}
```
and the caller side
```
val (cleanedOptions, path) = cleanTableProperties(ctx, options)
if (location.isDefined && path.isDefined) fail
```

Reply:
sgtm